### ISSUE TITLE
does text-generation-webui support Function Calling now?

### ISSUE THREAD

**Issue by Charles1757** [oobabooga/text-generation-webui#6539]
same as mentioned here, but I'm not sure if we are supporting this:
https://github.com/oobabooga/text-generation-webui/issues/4455

**Comment by Chepko932** [comment]
Same question, cant get it working so i guess no?

**Comment by jfmherokiller** [comment]
from the code based on the latest commit.  https://github.com/oobabooga/text-generation-webui/blob/4924ee2901ce9ad17259ae817bbb46885d1ba8ba/extensions/openai/completions.py#L221

```
    if body.get('function_call', ''):
        raise InvalidRequestError(message="function_call is not supported.", param='function_call')
```

**Comment by oobabooga** [comment]
Yes — tool/function calling is supported as of v4.0, for Qwen 3.5, Devstral 2, GPT-OSS, DeepSeek V3.2, GLM 5, MiniMax M2.5, Kimi K2/K2.5, and Llama 4. See the [release notes](https://github.com/oobabooga/text-generation-webui/releases/tag/v4.0) and [docs](https://github.com/oobabooga/text-generation-webui/wiki/12-%E2%80%90-OpenAI-API#tool-calling).
